integration-test/spark3/src/test/java/org/apache/uniffle/test/MockRssShuffleManager.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.google.common.collect.Maps;
+import org.apache.spark.SparkConf;
+import org.apache.spark.TaskContext;
+import org.apache.spark.shuffle.RssShuffleHandle;
+import org.apache.spark.shuffle.RssShuffleManager;
+import org.apache.spark.shuffle.ShuffleHandle;
+import org.apache.spark.shuffle.ShuffleReadMetricsReporter;
+import org.apache.spark.shuffle.ShuffleReader;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+import org.apache.uniffle.common.ShuffleServerInfo;
+
+public class MockRssShuffleManager extends RssShuffleManager {
```

Comment:
Why do we need this MockRssShuffleManager?

Reply:
> If the new version client access the old version server, what will happen? Could we fallback to use old way to access the old version server when there are exceptions?

I think the new version of the client does not need to be compatible with the old version of the server.

Reply:
Collect the number of times the partition block needs to be requested through MockRssShuffleManager#getReaderImpl, and then compare it with the appToPartitionRequest of MockedShuffleServerGrpcService to confirm that the number of requests is correct, and there are no redundant requests in the case of multiple replica, See `validateRequestCount(expectRequestNum * replicateRead)`

Reply:
So I think it's not a mock class. Could we give it a better name?

Reply:
If it is only bound to dedicated test cases, the name should be concrete. Or put this class into its dedicated cases.